server/src/main/java/org/opensearch/OpenSearchServerException.java

```diff
@@ -6,93 +6,37 @@
-import org.opensearch.cluster.action.shard.ShardStateAction;
-import org.opensearch.cluster.routing.NodeWeighedAwayException;
-import org.opensearch.cluster.routing.PreferenceBasedSearchNotAllowedException;
-import org.opensearch.cluster.routing.UnsupportedWeightedRoutingStateException;
-import org.opensearch.cluster.service.ClusterManagerThrottlingException;
-import org.opensearch.common.CheckedFunction;
-import org.opensearch.common.collect.Tuple;
-import org.opensearch.common.io.stream.StreamInput;
-import org.opensearch.common.io.stream.StreamOutput;
-import org.opensearch.common.io.stream.Writeable;
-import org.opensearch.core.xcontent.XContentBuilder;
-import org.opensearch.core.xcontent.XContentParser;
-import org.opensearch.index.Index;
-import org.opensearch.index.shard.ShardId;
-import org.opensearch.rest.RestStatus;
-import org.opensearch.search.aggregations.MultiBucketConsumerService;
-import org.opensearch.search.pipeline.SearchPipelineProcessingException;
-import org.opensearch.snapshots.SnapshotInUseDeletionException;
-import org.opensearch.transport.TcpTransport;
+import org.opensearch.index.snapshots.IndexShardSnapshotException;
 
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import static org.opensearch.BaseOpenSearchException.OpenSearchExceptionHandleRegistry.registerExceptionHandle;
+import static org.opensearch.OpenSearchException.OpenSearchExceptionHandle;
+import static org.opensearch.OpenSearchException.OpenSearchExceptionHandleRegistry.registerExceptionHandle;
+import static org.opensearch.OpenSearchException.UNKNOWN_VERSION_ADDED;
 import static org.opensearch.Version.V_2_1_0;
 import static org.opensearch.Version.V_2_4_0;
 import static org.opensearch.Version.V_2_5_0;
 import static org.opensearch.Version.V_2_6_0;
 import static org.opensearch.Version.V_2_7_0;
 import static org.opensearch.Version.V_3_0_0;
-import static org.opensearch.cluster.metadata.IndexMetadata.INDEX_UUID_NA_VALUE;
-import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
-import static org.opensearch.common.xcontent.XContentParserUtils.ensureFieldName;
 
 /**
- * A base class for all opensearch exceptions.
+ * Utility class to register server exceptions
  *
  * @opensearch.internal
  */
-public class OpenSearchException extends BaseOpenSearchException implements Writeable {
+public final class OpenSearchServerException {
+
+    private OpenSearchServerException() {
+        // no ctor:
+    }
 
     /**
      * Setting a higher base exception id to avoid conflicts.
      */
     private static final int CUSTOM_ELASTICSEARCH_EXCEPTIONS_BASE_ID = 10000;
 
-    static {
-        registerExceptionHandle(
-            new OpenSearchExceptionHandle(
-                org.opensearch.index.snapshots.IndexShardSnapshotFailedException.class,
```

Comment:
Should the rest of the snapshot exceptions in this class get moved?  I don't think they are server specific. Or were you doing a couple here `IndexShardSnapshotException` and `IndexShardSnapshotFailedException` as a starting point?

InvalidSnapshotNameException
SnapshotInProgressException
SnapshotInUseDeletionException

Also on that note all the shard level exceptions like `IndexShardStartedException` to libs/core

Reply:
Starting point to demonstrate the registration. The rest of them will move in a follow up PR. This minimizes the change surface area.